### PR TITLE
Fyle-25g66c: Currency auto-fill in expense page

### PR DESCRIPTION
**Pull request by Dimple16**

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -788,7 +791,8 @@ export class AddEditExpensePage implements OnInit {
       eou: eou$,
       instaFyleSettings: instaFyleSettings$,
       imageData: this.getInstaFyleImageData(),
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:793]: To be reviewed from here:

**Comment by Dimple16** [comment]
https://user-images.githubusercontent.com/31147415/109160735-30aa4200-779c-11eb-9512-cba7f0c7f6e3.mov

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 2]
```diff
@@ -64,7 +64,6 @@ import {RecentlyUsed} from 'src/app/core/models/recently_used.model';
 export class AddEditExpensePage implements OnInit {
   etxn$: Observable<any>;
   paymentModes$: Observable<any[]>;
-  pickRecentCurrency$: Observable<any>;
```
Comment (iamtarun1993) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:67]: Make sure we haven't used this variable in the HTML page

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -792,7 +791,8 @@ export class AddEditExpensePage implements OnInit {
       eou: eou$,
       instaFyleSettings: instaFyleSettings$,
       imageData: this.getInstaFyleImageData(),
-      recentCurrency: from(this.recentLocalStorageItemsService.get('recent-currency-cache'))
+      recentCurrency: from(this.recentLocalStorageItemsService.get('recent-currency-cache')),
+      recentValue: this.recentlyUsedValues$
```
Comment (iamtarun1993) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:795]: where we declared and defined `recentlyUsedValues$`..
can't find it in this PR. 
Is it in another PR??
can you put the other PR link over here??

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -834,6 +835,9 @@ export class AddEditExpensePage implements OnInit {
             if (orgUserSettings.currency_settings.preferred_currency) {
               etxn.tx.currency = orgUserSettings.currency_settings.preferred_currency;
             }
+          } else if (orgUserSettings.expense_form_autofills && orgUserSettings.expense_form_autofills.allowed && orgUserSettings.expense_form_autofills.enabled
```
Comment (iamtarun1993) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:838]: Please add `expense_form_autofills` in orgUserSettings model

Comment (iamtarun1993) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:838]: Then no need to do
`orgUserSettings.expense_form_autofills && orgUserSettings.expense_form_autofills.allowed && orgUserSettings.expense_form_autofills.enabled`

only 
`orgUserSettings.expense_form_autofills.allowed && orgUserSettings.expense_form_autofills.enabled` will be enough

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:795]: added in this pr - https://github.com/fylein/fyle-mobile-app2/pull/239

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:67]: yes, checked this

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 3 above)
Comment (iamtarun1993) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:795]: This is ok

File: src/app/core/models/org_user_settings.model.ts [hunk 5]
```diff
@@ -116,6 +116,11 @@ export interface InAppChatSettings {
     restore_id?: any;
 }
 
+export interface ExpenseFormAutofills {
+    allowed: boolean;
```
Comment (iamtarun1993) [inline src/app/core/models/org_user_settings.model.ts:120]: 2 space tab

**Review by iamtarun1993 — APPROVED** [review]

File: src/app/core/models/org_user_settings.model.ts (hunk 5 above)
Comment (abhishek1234321) [reply src/app/core/models/org_user_settings.model.ts:120]: +1

**Review by abhishek1234321 — APPROVED** [review]

**Pull request closed without merging**